### PR TITLE
Send provider email when request is created

### PR DESCRIPTION
**Pull request by tuntisz**
https://github.com/cantino/feedadoc/issues/11

When a provider creates a request, we send them an email with their request's public link and an edit/delete link.
Screenshot_2020-03-21 Screenshot

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -32,6 +32,10 @@
   config.action_mailer.raise_delivery_errors = false
 
   config.action_mailer.perform_caching = false
+  config.action_mailer.raise_delivery_errors = false
+
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.smtp_settings = { address: 'mailcatcher', port: 1025, domain: 'mailcatcher' }
```
Comment (tuntisz) [inline config/environments/development.rb:38]: This change requires folks to map `mailcatcher` to localhost in `/etc/hosts/` if wanting to see email in mailcatcher. Without doing so, nothing breaks, you just won't be able to verify email sends. This has been added in the README.

File: app/graphql/mutations/create_provider_and_request.rb [hunk 2]
```diff
@@ -40,7 +40,9 @@ def resolve(first_name:, last_name: "", neighborhood: "", city:, state:, email:,
     if provider.valid? && request.valid?
       provider.save!
       request.save!
-      LinkCreator.create(request)
+      linked_token = LinkCreator.create_token(request)
+      # TODO: Do we plan on using Action Job? If so, make async
```
Comment (cantino) [inline app/graphql/mutations/create_provider_and_request.rb:44]: Inline is likely okay for now; we can make a story to move to background.

File: app/mailers/provider_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class ProviderMailer < ApplicationMailer
+  def request_created_email
+    @provider = params[:linked_token].entity.provider
+    @request_url = 'localhost:3000/' + 'request/' + params[:linked_token].entity.id.to_s
```
Comment (cantino) [inline app/mailers/provider_mailer.rb:4]: localhost:3000 won't work in the final system right. I think we need [something like this](https://stackoverflow.com/questions/798917/how-do-i-configure-the-hostname-for-rails-actionmailer)?

Comment (dpix) [inline app/mailers/provider_mailer.rb:4]: Probably going to need a way to make this base url configurable?

Or can you get the current domain from the request somehow?

File: docker-compose.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ services:
   web:
     build: .
     container_name: feedadoc_web
-    command: bash -c "bundle && rm -f tmp/pids/server.pid && bundle exec rails s -p 3000 -b '0.0.0.0'"
+    command: tail -f /dev/null
```
Comment (cantino) [inline docker-compose.yml:17]: What is this doing?

File: app/mailers/provider_mailer.rb (hunk 3 above)
Comment (tuntisz) [reply app/mailers/provider_mailer.rb:4]: I _think_ this comment might be misplaced but if not, this is the URL that will route to viewing a request to be shared on social.

As for actionmailer config, I think I only edited the development config. I can do production config in the sendgrid setup work.

File: docker-compose.yml (hunk 4 above)
Comment (tuntisz) [reply docker-compose.yml:17]: When the Docker command was the rails server, terminating the server would stop the container. By switching to this, we can use scripts to start/stop/restart. 

The issue this solved for me was the fact that emails wouldn't reload the code (and I didn't yet know about rails email preview).  So in order to verify my changes, I was having to stop and restart the docker container which was a slow feedback loop

File: app/mailers/provider_mailer.rb (hunk 3 above)
Comment (tuntisz) [reply app/mailers/provider_mailer.rb:4]: Good point! I was lazy here as I'm not too familiar with how Rails environment variables work and I wasn't sure if the app would be hosted on feedadoc.com or some dev iteration yet. I'll convert url definition into some sort of service and maybe we can circle back to update with environment vars later? Unless someone wants to school me on what that looks like in Rails land.

**Comment by cantino** [comment]
Looks good to me, go ahead and merge!

**Review by cantino — APPROVED** [review]

File: app/mailers/provider_mailer.rb (hunk 3 above)
Comment (tuntisz) [reply app/mailers/provider_mailer.rb:4]: Just to wrap this conversation, I'm going to merge as is and circle back to this elsewhere. Great call out, though. Thanks!

**Pull request merged**